cmd/controller/app/options/options.go

```diff
@@ -200,9 +206,22 @@ func (s *ControllerOptions) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&s.DefaultACMEIssuerDNS01ProviderName, "default-acme-issuer-dns01-provider-name", defaultACMEIssuerDNS01ProviderName, ""+
 		"Required if --default-acme-issuer-challenge-type is set to dns01. The DNS01 provider to use for ingresses using ACME dns01 "+
 		"validation that do not explicitly state a dns provider.")
-	fs.StringSliceVar(&s.DNS01Nameservers, "dns01-self-check-nameservers", []string{}, ""+
-		"A list of comma seperated DNS server endpoints used for DNS01 check requests. "+
-		"This should be a list containing IP address and port, for example: 8.8.8.8:53,8.8.4.4:53")
+	fs.StringSliceVar(&s.DNS01RecursiveNameservers, "dns01-recursive-nameservers",
+		[]string{}, "A list of comma seperated dns server endpoints used for "+
+			"DNS01 check requests. This should be a list containing IP address and "+
+			"port, for example 8.8.8.8:53,8.8.4.4:53")
+	fs.BoolVar(&s.DNS01RecursiveNameserversOnly, "dns01-recursive-nameservers-only",
+		defaultDNS01RecursiveNameserversOnly,
+		"When true, cert-manager will only ever query the configured DNS resolvers "+
+			"to perform the ACME DNS01 self check. This is useful in DNS constrained "+
+			"environments, where access to authoritative nameservers is restricted. "+
+			"Enabling this option could cause the DNS01 self check to take longer "+
+			"due to caching performed by the recursive nameservers.")
+	fs.StringSliceVar(&s.DNS01RecursiveNameservers, "dns01-self-check-nameservers",
```

Comment:
Is it possible to have both flags reference the same string/ptr location? It seems like these two would flip-flop and both try and set different values?

We've not handled deprecating flags yet - I was envisioning we just don't introduce the new flag name for now (and defer the rename for when we move to a Config API type structure). If it's easy to deprecate, then great 👍 if not, I think we can defer adding the new `--dns01-recursive-nameservers` field name until the move to Config API types 😄

Reply:
You can specify both but the last one used will win. So --test1 --test2 where they both use the same var test2 will win out and vice versa for the reverse.

I can remove the flag change. I thought do it while I am there so it doesn't become confusing for a new comer. The old flag will still keep working just spit our a warning on start to swap over to the new one.

Let me know what the desired outcome is.

Cheers

Reply:
Great, thanks for doing the investigation! If that's the case, +1 to keeping it as-is in this PR and deprecating the old flag 😄